specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/AppServiceEnvironments.json

```diff
@@ -1122,6 +1122,60 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/hostingEnvironments/{name}/upgrade": {
+      "post": {
+        "tags": [
+          "AppServiceEnvironments"
+        ],
+        "summary": "Initiate an upgrade of an App Service Environment if one is available.",
+        "description": "Description for Initiate an upgrade of an App Service Environment if one is available.",
+        "operationId": "AppServiceEnvironments_Upgrade",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the App Service Environment.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "testNotification",
+            "in": "query",
+            "description": "Whether the request is to send an upgrade test notification instead of starting an upgrade.",
+            "type": "boolean"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
>200 [](http://example.com/codeflow?start=11&length=3)

ARM requires all APIs to be either always sync or always async. Since you are supporting async, you don't need sync status codes.